src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -89,3 +95,19 @@ void Qgs3DMapCanvas::setViewFromTop( const QgsPointXY &center, float distance, f
   float worldY = center.y() - mMap->origin().y();
   mScene->cameraController()->setViewFromTop( worldX, -worldY, distance, rotation );
 }
+
+void Qgs3DMapCanvas::saveAsImage( const QString fileName, const QString fileFormat )
+{
+  if ( !fileName.isEmpty() )
+  {
+    mCaptureReply = mCapture->requestCapture();
+    connect( mCaptureReply, &Qt3DRender::QRenderCaptureReply::completed, this, [ = ]
+    {
+      mCaptureReply->image().save( fileName, fileFormat.toLocal8Bit().data() );
+      emit savedAsImage( fileName );
+
+      delete mCaptureReply;
```

Comment:
It would be safer to use deleteLater() here: from what I remember in slots one should not delete the object which triggered the slot.

Reply:
OK.